src/main/java/com/testlio/plugins/jenkins/ScheduleAutomatedRunBuilder.java

```diff
@@ -13,318 +29,407 @@
-            throws InterruptedException, IOException {
-        listener.getLogger().println(this);
-
-        RunConfigurationAction config = new RunConfigurationAction(
-                projectId,
-                testRunCollectionGuid,
-                automatedTestRunCollectionGuid,
-                appType,
-                testPackageURI,
-                videoCapture,
-                select,
-                waitForResults
-        );
-
-        if (isBrowser()) {
-            config.setBrowserTestType(browserTestType);
-            config.setBrowserSelector(browserSelector);
-            config.setTimeoutBrowsers(timeoutBrowsers);
-        } else {
-            config.setAppBuildURI(appBuildURI);
-            config.setDeviceTestType(deviceTestType);
-            config.setDeviceSelector(deviceSelector);
-            config.setDeviceState(deviceState);
-            config.setTimeoutDevices(timeoutDevices);
-        }
-
-        if (testArgs != null) config.setTestArgs(testArgs);
-
-        run.addAction(config);
-
-        // TODO Implement automated run scheduling logic here
-        listener.getLogger().println("Plugin works!");
+    listener.getLogger().println("Starting process to create an automation run...");
+    BrowsersDTO browsersDTO = null;
+    DevicesDTO devicesDTO = null;
+    if (isBrowser()) {
+      listener.getLogger().println("Step 0: Search browsers");
+      browsersDTO = searchBrowsers(config, restClient);
+    }
+    else {
+      listener.getLogger().println("Step 0: Search devices");
+      devicesDTO = searchDevices(config, restClient);
+    }
+
+    listener.getLogger().println("Step 1: Create a new automated run");
+    RunDTO automatedRun = createNewAutomatedRun(listener, restClient, config);
+
+    listener.getLogger().println("Step 2: Create and Update automated run configuration");
+    createAndUpdateAutomatedRunConfiguration(listener, restClient, automatedRun, config, isBrowser());
+
+    listener.getLogger().println("Step 3: Create automated run plan");
+    ResponseHrefDTO automatedTestPlan = (ResponseHrefDTO) restClient.post(automatedRun.getPlans().getHref(), new JSONObject(), ResponseHrefDTO.class);
+
+    listener.getLogger().println("Step 4: Upload test build");
+    uploadBuild(listener, restClient, buildFile, automatedTestPlan, isTestTypeDevice(), config, isIOS());
+
+    listener.getLogger().println("Step 5: Upload test package");
+    uploadTestPackage(listener, restClient, testPackageFile, automatedTestPlan, config);
+
+    listener.getLogger().println("Step 6: Devices");
+    attachDevices(listener, restClient, browsersDTO, devicesDTO, automatedTestPlan, config, isBrowser());
+
+    listener.getLogger().println("Step 7. Schedule run! 🚀");
+    scheduleAutomatedRun(listener, restClient, automatedRun, config);
+
+    listener.getLogger().println("Step 8. Wait for results");
+    pollResults(listener, restClient, automatedRun, config);
```

Comment:
Does `scheduleAutomatedRun` throw an error if the HTTP response isn't 200? In [testlio-cli](https://github.com/Testlio/testlio-cli/blob/3b1c24913203e2412390ed0dbc80352e0327015c/bin/create-run.js#L528) we have a check to only begin polling if the response was 200.

Reply:
Incoporated, if response is not 200 it will throw error